src/jsonrpc/api.rs

```diff
@@ -226,6 +226,34 @@ fn list_confirmed(control: &DaemonControl, params: Params) -> Result<serde_json:
     ))
 }
 
+fn list_spendtxs(
+    control: &DaemonControl,
+    params: Option<Params>,
+) -> Result<serde_json::Value, Error> {
+    let txids: Option<Vec<bitcoin::Txid>> = if let Some(p) = params {
+        let tx_ids = p.get(0, "txids");
+        if let Some(ids) = tx_ids {
+            let ids: Vec<Txid> = ids
+                .as_array()
+                .and_then(|arr| {
+                    arr.iter()
+                        .map(|entry| entry.as_str().and_then(|e| bitcoin::Txid::from_str(e).ok()))
+                        .collect()
+                })
+                .ok_or_else(|| Error::invalid_params("Invalid 'txids' parameter."))?;
+            if ids.is_empty() {
+                return Err(Error::invalid_params("Invalid 'txids' parameter."));
+            }
```

Comment:
This check should be present for both interfaces (through the crate and through the JSONRPC), not only the JSONRPC.

Reply:
i've moved the check to the command level, created `CommandError::EmptyFilterList` ,a quite generalistic name in case we want to raise it later for other filter list, but can rename it to `CommandError::EmptyTxIdsList` if you want something more specific.